custom_components/energidataservice/tariffs/energidataservice/__init__.py

```diff
@@ -159,3 +147,27 @@ async def async_get_system_tariffs(self) -> dict:
                     )
 
         self._additional_tariff = tariff_data
+
+    @retry(attempts=5)
+    async def async_call_api(self, query: str) -> dict:
+        """Make the API calls."""
+        try:
+            headers = self._header()
```

Comment:
You should probably introduce a wait here not to spam the API. Potentially with exponential backoff. Most likely something like `await asyncio.sleep(0.6)`

Reply:
Also, since many people use this plugin, this gives a huge peak at the API provided. Consider adding an artificial random wait to spread the load for the receiving API provider.

Reply:
> Also, since many people use this plugin, this gives a huge peak at the API provided. Consider adding an artificial random wait to spread the load for the receiving API provider.

Already have that, it follows the random refresh time for the spot price